cmd/catalog-importer/cmd/sync.go

```diff
@@ -281,19 +321,65 @@ func (opt *SyncOptions) Run(ctx context.Context, logger kitlog.Logger, cfg *conf
 
 				version := result.JSON200.CatalogType.Schema.Version
 				logger.Log("msg", "updating catalog type schema", "catalog_type_id", catalogType.Id, "version", version)
-				schemaResult, err := cl.CatalogV2UpdateTypeSchemaWithResponse(ctx, catalogType.Id, client.CatalogV2UpdateTypeSchemaJSONRequestBody{
+				schema, err := cl.CatalogV2UpdateTypeSchemaWithResponse(ctx, catalogType.Id, client.CatalogV2UpdateTypeSchemaJSONRequestBody{
 					Version:    version,
-					Attributes: model.Attributes,
+					Attributes: attributesWithoutNewBacklinks,
 				})
 				if err != nil {
 					return errors.Wrap(err, "updating catalog type schema")
 				}
 
-				updatedCatalogType = schemaResult.JSON200.CatalogType
+				catalogTypeVersions[catalogType.Id] = schema.JSON200.CatalogType.Schema.Version
+
+				OUT("  ✔ %s (id=%s)", model.TypeName, catalogType.Id)
 			}
-			OUT("  ✔ %s (id=%s)", model.TypeName, catalogType.Id)
-			if opt.DryRun {
-				DIFF("  ", *catalogType, updatedCatalogType)
+		}
+
+		// Then go through again and create any types that do have new backlinks
+		OUT("\n↻ Syncing backlink attributes...")
+		for _, outputType := range cfg.Outputs() {
+			baseModel, enumModels := output.MarshalType(outputType)
+			for _, model := range append(enumModels, baseModel) {
+				catalogType := catalogTypesByOutput[model.TypeName]
+
+				hasNewBacklinks := false
+				for _, attr := range model.Attributes {
+					if attr.Mode != nil && attr.BacklinkAttribute != nil {
+						inCurrentSchema := false
+						// Does it exist in the current schema?
+						if attr.Id == nil {
+							logger.Log("msg", "attribute ID is empty, which shouldn't be possible")
+							continue
+						}
+
+						for _, existingAttr := range catalogType.Schema.Attributes {
+							if existingAttr.Id == *attr.Id {
+								inCurrentSchema = true
+								break
+							}
```

Comment:
similar to above, this can be simpler imo
```suggestion
							if existingAttr.Id != *attr.Id {
								hasNewBacklinks = true
								break // Break the outer loop here
							}
```

Reply:
I don't think this works - I need to check every attribute to know if my attribute was there, otherwise I'd bail once I found the first attribute that didn't match

Reply:
I've just realised I can use lo, gonna do that as its way more readable

Reply:
Ah you're right, my bad. lol that was my first thought and I was like "huh I guess we're not using lo in this repo for some reason" 😂